### ISSUE TITLE
[Question] VS Code generates `import { observable } from "mobx/lib/api/observable";`

### ISSUE THREAD

**Issue by chenyong** [mobxjs/mobx#1276]
VS Code `Version 1.18.1 (1.18.1)` and `"mobx": "^3.1.16"`.

VS Code automatically infers my code `observable` and generates `import` statement for me. However, the generated code is:

```js
import { observable } from "mobx/lib/api/observable";
```

which should be:

```js
import { observable } from "mobx";
```

Anything we can do to fix it?

**Comment by mweststrate** [comment]
Yes it is super annoying, but no idea how to fix it. It might be interesting to bring it up in the VSCode issue tracker, as the problem is imho that the typings are imported from the originally declaring file of the typings (`lib/api/observable.d.ts`), rather than the file that finally exports and exposes it to the outside world (`index.d.ts`).

The problem is that the first does match an actual javascript file, but that is only if you are consuming it like common js. However, any bundler will pick the es module instead which is located differently inside the package, but to whom the same typings apply. So effectively the first import causes the bundler to use the common .js files, and the latter to use the most appropriate (the ES module build usually) as declared in the `package.json` of mobx.

**Comment by mweststrate** [comment]
See also https://github.com/Microsoft/TypeScript/issues/19694  / https://github.com/Microsoft/TypeScript/issues/19694, this should be fixed with the upcoming Typescript 2.7 release

**Comment by chenyong** [comment]
These two links are identical...
